proposals/002-text-utf-default.md

```diff
@@ -0,0 +1,156 @@
+
+Using the HVR's existing [`text-utf8`](https://github.com/text-utf8) as
+a starting point, the following must be done before an implementation is
+started:
+
+-   Modernize the codebase and clear out the bitrot
+
+-   Establish a baseline for performance and any related issues.
+
+-   Update testing and performance benchmarks to make use of `inspection-testing` to ensure fusion is not broken in
+    subsequent UTF-8 related changes.
+
+An MVP should completely preserve standard user-facing API, and not
+break fusion. Performance should not significantly diverge from the
+existing UTF-16 text package. There will be an expected change to the
+exposed Text internals, in which case, breakage should be assessed by
+circulating a git commit reference to a release candidate as soon as
+possible. This candidate should be sourced publicly and loudly.
+
+**Implementation:**
+
+-   TBD: There is a straightforward implementation, but this one is left up to Andrew for comment.
+
+**Stakeholders:**
+
+-   Library authors will need to be made aware of changes and adjust accordingly. HF will provide a git reference to a complete MVP as
+    soon as possible, and produce a migration guide.
+
+-   In the case of GHC, Cabal, and other core infrastructure, we will work closely with these packages to help migrate and assess+fix
+    breakages.
+
+While we do not expect many authors to experience significant changes,
+there will be some help that needs to be given in terms of bumping
+Hackage bounds since this migration will be a major version bump. HF
+will need to coordinate with the Hackage Trustees to help move along
+packages that go out of date.
+
+# Deliverables
+
+-   text-2.0.0.0, which will provide a UTF-8 encoding for Text as a default for all versions going forward.
+
+-   A `text-utf16` package, which is a preservation of the current UTF-16 encoded text, for backwards compatibility.
+
+-   Updates to the Text Haddocks that reflect the UTF-8 changes
+
+-   Announcements and updates across all Haskell channels covering the following:
+    -   Significant dates and milestones
+
+    -   Expected code impact
+
+    -   Release candidates
+
+    -   Delivery
+
+-   Migration instructions and design documentation
+
+# Outcomes
+
+-   Addresses a recognized need and want from both Industry and the Haskell Community
+
+-   Better UX for Haskell's text story
+
+-   Establishes the ability for Haskell Foundation to Get Things Done that were previously blocked for 10+ years in the community.
+
+-   Better interoperability and web story
+
+# Risks
+
+-   HF must minimize the cost to migrate, or people will just get mad (and rightfully so).
+
+-   Text-icu will need a bespoke UTF-8 conversion function. In general, the Unicode story must be tracked and made sure it will not break.
+    -   Recommendation: make this a high-priority deliverable when project planning
+
+-   The old UTF-16 text package will need to be preserved, and will require a maintainer.
```

Comment:
Why?

Reply:
I also wonder why. Certainly libraries that use `text` internals (like `hashable`) will need to be re-written or have a new CPP case to handle the old encoding, but I don't immediately see the utility of `old-text`

Reply:
Why? Because there are many cases in which utf-16 _is_ the better choice -- that's why it was picked in the first place!

Reply:
Yes, this is true but In this case we should call the existing package not `old-text` but rather `text-utf16`.

Reply:
Why the qualifier? I see no reason why someone shouldn't be able to pick up `text-utf16` and continue maintaining it as the canonical UTF-16 text package. We don't want it to be the default, or perhaps be in `base`, but it's not a _bad_ package per se, as is implied by `old`.

Reply:
I believe that supporting `text-utf16` is up to community. I mean, it's trivial to create such a package: `cabal unpack text-1.2.4.1; cd text-1.2.4.1; mv text.cabal text-utf16.cabal; sed 's/text/text-utf16/g' test-utf16.cabal; cabal sdist; cabal upload`, anyone can do it in a minute. We can put a call for volunteers and bless them to proceed with a fork.

Reply:
That's a fair split i think @Bodigrim